### PR TITLE
gimp: condition python2 behind new withPython argument

### PR DESCRIPTION
**Pull request by fabianhjr**
###### Description of changes

Conditionally Python2 dep (EOLed 2020-01-01) as per #201859

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
> I wonder how important the python stuff is in gimp. The security issues in python2 will get fixed in nixpkgs (probably), so it might be OK to keep it here (by default).

I think we need to add it to the release notes for 23.05 at least.

Looking at https://bugzilla.redhat.com/show_bug.cgi?id=1737933, Python looks like necessary for a good part of the core functionality of Gimp, so yeah, disabling by default may be bad.

However, I would still go with this as-is. While I am trying to get Python 2 in a good state here in nixpkgs for the use cases that I think it matters (mainly for old packages that will never be updated to work with Python 3), if we can reduce the users of Python 2, it is better.

IMO, we should work in adding Gimp 2.99.x as an option for those that want it. Gimp 2.99.x will bring both GTK3 and Python 3 support, so this would fix two of our main deprecation targets in one package.

**Comment by fabianhjr** [comment]
Afaik it is used for plugins and not as common.

@jtobin might have better information.

**Comment by thiagokokada** [comment]
> Afaik it is used for plugins and not as common.
> 
> @jtobin might have better information.

Arch Linux seems to ship it without support for Python, so this looks good: https://github.com/archlinux/svntogit-packages/blob/d25b5ac3e107623411b25ffde09cd74efc37c457/trunk/PKGBUILD#L50

File: pkgs/applications/graphics/gimp/default.nix [hunk 1]
```diff
@@ -116,9 +117,6 @@ in stdenv.mkDerivation rec {
     shared-mime-info
     libwebp
     libheif
-    python
-    # Duplicated here because python.withPackages does not expose the dev output with pkg-config files
```
Comment (dotlambda) [inline pkgs/applications/graphics/gimp/default.nix:120]: Can we keep this comment?

Comment (fabianhjr) [reply pkgs/applications/graphics/gimp/default.nix:120]: Woops, yeah. My mistake. Amending

Comment (fabianhjr) [reply pkgs/applications/graphics/gimp/default.nix:120]: Amended

**Review by dotlambda — APPROVED** [review]

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 203395` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gimpPlugins.exposureBlend</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimp-with-plugins</li>
    <li>gimpPlugins.bimp</li>
    <li>gimpPlugins.farbfeld</li>
    <li>gimpPlugins.fourier</li>
    <li>gimpPlugins.gap</li>
    <li>gimpPlugins.gimplensfun</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.lqrPlugin</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>gimpPlugins.texturize</li>
    <li>gimpPlugins.waveletSharpen</li>
    <li>toppler</li>
  </ul>
</details>

File: pkgs/applications/graphics/gimp/default.nix [hunk 2]
```diff
@@ -48,6 +47,8 @@
 , AppKit
 , Cocoa
 , gtk-mac-integration-gtk2
+, withPython ? false
+, python2 ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/gimp/default.nix:51]: Why do we need the ? null here? We are not going to drop the python2 package from the tree.

Comment (dotlambda) [reply pkgs/applications/graphics/gimp/default.nix:51]: We might in the future.

Comment (SuperSandro2000) [reply pkgs/applications/graphics/gimp/default.nix:51]: Then we can still do this in the future or just remove the dead code.

**Comment by adisbladis** [comment]
This is a breaking change and should at least come with a release note.

**Pull request merged**